test/mock_test.py

```diff
@@ -42,6 +43,30 @@ def test_with(self):
         with t.open('r') as b:
             self.assertEqual(list(b), ['bar'])
 
+    def test_bytes(self):
+        t = MockTarget("foo", format=Nop)
+        with t.open('wb') as b:
+            b.write(b"bar")
+
+        with t.open('rb') as b:
+            self.assertEqual(list(b), [b'bar'])
+
+    def test_default_mode_value(self):
+        t = MockTarget("foo")
+        with t.open('w') as b:
+            b.write("bar")
+
+        with t.open() as b:
+            self.assertEqual(list(b), ['bar'])
+
+    def test_mode_none_error(self):
+        t = MockTarget("foo")
+        try:
+            with t.open(None) as b:
+                b.write("bar")
+        except TypeError:
+            self.assertRaises(TypeError)
```

Comment:
with this being in an `except`, we can't guarantee that the assertion is risen

i was expecting

```python
t = MockTarget("foo")
with self.assertRaises(TypeError):
    with t.open(None) as b:
        b.write("bar")
```

Reply:
Ok, going to refactor it then